# Metabolic Modelling Course

## Trainers
Mary-Ann Blätke (<blaetke@ipk-gatersleben.de>)

Jedrzej Jakub Szymanski (<szymanski@ipk-gatersleben.de>)

[HP](https://www.ipk-gatersleben.de/molekulare-genetik/netzwerkanalyse-und-modellierung/)

## Part 1 - Introduction
In the first part of this course, we will discuss the importance of mathematical modelling to enhance the understanding of real-world systems and to make predictions about the system's behaviour. 
Next, we will focus on metabolic networks, their construction and purpose and the application of flux balance analysis to investigate the flux distribution optimised for specific criteria subjected to a set of constraints.

### References
* Orth, J. D., Thiele, I., & Palsson, B. Ø. (2010). What is flux balance analysis?. Nature biotechnology, 28(3), 245. [DOI](https://doi.org/10.1038/nbt.1614)
* Lewis, N. E., Nagarajan, H., & Palsson, B. O. (2012). Constraining the metabolic genotype–phenotype relationship using a phylogeny of in silico methods. Nature Reviews Microbiology, 10(4), 291. [DOI](https://doi.org/10.1038/nrmicro2737)
* Feist, A. M., & Palsson, B. Ø. (2008). The growing scope of applications of genome-scale metabolic reconstructions using Escherichia coli. Nature biotechnology, 26(6), 659. [DOI](https://doi.org/10.1038/nbt1401)
* Terzer, M., Maynard, N. D., Covert, M. W., & Stelling, J. (2009). Genome‐scale metabolic networks. Wiley Interdisciplinary Reviews: Systems Biology and Medicine, 1(3), 285-297. [DOI](https://doi.org/10.1002/wsbm.37)

### Other Resources
* [USU course on "Systems Biology: Constraint-based Metabolic Reconstructions and Analysis"](https://systemsbiology.usu.edu/introduction.php)

## Part 2 - Exercise
In the second part of this course, we will go through the [cobrapy](https://opencobra.github.io/cobrapy/) Tutorial provided at [ReadTheDocs](https://cobrapy.readthedocs.io/en/latest/index.html) to learn about the very basics, from loading to manipulation/building models to perform a simple FBA and beyond.

### Working Environment
We use the [python](https://www.python.org) as a programming language and the [cobrapy](https://opencobra.github.io/cobrapy/) package for constraint-based modelling of biological networks, which is part of the [openCobra](http://opencobra.github.io) project. 
For live coding, we will use the [Jupyter Notebook](https://jupyter.org) implementation at [Google Colabratory](https://colab.research.google.com/). 
Jupyter Notebook is an open-source web application that allows you to create and share documents that contain live code, equations, visualisations and narrative text. If you prefer to run your own Jupyter Notebook instance, we recommend installing it via [Anaconda](https://anaconda.org).

### Requirements
Some experience in coding, preferably using python. 
Resources for learning python can be found at:
* [Codecademy](https://www.codecademy.com/catalog/language/python)
* [Datacamp](https://www.datacamp.com/courses/q:python)
* [Udemy](https://www.udemy.com/topic/python/)
* [Learn Python](https://www.learnpython.org)
* [Programiz](https://www.programiz.com/python-programming#tutorial)
* [More...](https://wiki.python.org/moin/BeginnersGuide/Programmers)

You might also find these [Cheat Sheets](https://www.datacamp.com/community/data-science-cheatsheets?tag=python) helpful if you just need a refresher.
